ambry-api/src/main/java/com.github.ambry/clustermap/ClusterMap.java

```diff
@@ -32,15 +32,20 @@
   PartitionId getPartitionIdFromStream(InputStream stream) throws IOException;
 
   /**
-   * Gets a list of partitions that are available for writes.
+   * Gets a list of partitions that are available for writes. The returned list can be manipulated.
+   * @param partitionClass the partition class whose writable partitions are required. Can be {@code null}
+   * @return a list of all writable partitions belonging to the partition class (or all writable partitions if
+   * {@code partitionClass} is {@code null})
    */
-  List<? extends PartitionId> getWritablePartitionIds();
+  List<? extends PartitionId> getWritablePartitionIds(String partitionClass);
 
   /**
-   * Gets a list of all partitions in the cluster
-   * @return a list of all partitions in the cluster
+   * Gets a list of all partitions in the cluster. The returned list can be manipulated.
+   * @param partitionClass the partition class whose partitions are required. Can be {@code null}
+   * @return a list of all partitions belonging to the partition class (or all partitions if {@code partitionClass} is
+   * {@code null})
    */
-  List<? extends PartitionId> getAllPartitionIds();
+  List<? extends PartitionId> getAllPartitionIds(String partitionClass);
```

Comment:
- Cannot be an enum because the partition classes can't be (and need not be) exhaustively listed in code
- I don't want the interface to get repetitive. In reality, only tests send in `null`. All production code should send in an actual string (after the PR where the container config dictates the partition class)